### PR TITLE
[Doc] peregrine targets ref docs

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

This PR updates the JSDocs in the target intercept and declare files in the Peregrine package.
It also creates a file in devdocs that pulls this information into a reference-style topic.

## Related Issue

Closes PWA-681

## Acceptance 

@zetlen 

### Verification Stakeholders

@zetlen 

### Verification Steps
1. Navigate to the pwa devdocs directory: `cd pwa-devdocs`
1. Start the HTML preview server: `yarn develop`
1. Verify files are auto-generated:

    - `src/_includes/auto-generated/peregrine/lib/targets/peregrine-intercept.md`
    - `src/_includes/auto-generated/peregrine/lib/targets/peregrine-declare.md`

1. On the devdocs preview site, in the top navigation, click on **Peregrine API** under the _Reference API_ item.
1. Verify the Extensibility Targets topic appears with content.
1. Verify this topic is listed on the left navigation for this section
1. Verify the Wrappable Talons topic is listed on the left navigation and links to the correct page 

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2492.pwa-venia.com : LH Performance Expected 0.85 Actual 0.52, LH Best Practices Expected 1 Actual 0.92
https://pr-2492.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.34, LH Best Practices Expected 1 Actual 0.92
https://pr-2492.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.49, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92

File: packages/peregrine/lib/targets/peregrine-intercept.js [hunk 1]
```diff
@@ -24,17 +41,27 @@ function TalonWrapperConfig(addTransforms) {
     });
     return {
         /**
-         * @memberof TalonWrapperConfig
+         * Provides access to the talon used in Venia's ProductFullDetail UI component.
```
Comment (jimbo) [inline packages/peregrine/lib/targets/peregrine-intercept.js:44]: In general, I think `component` is sufficient.

```suggestion
         * Provides access to the talon used in Venia's ProductFullDetail component.
```

File: packages/peregrine/lib/targets/peregrine-intercept.js [hunk 2]
```diff
@@ -24,17 +41,27 @@ function TalonWrapperConfig(addTransforms) {
...
                 'ProductFullDetail/useProductFullDetail',
                 'useProductFullDetail'
             )
         },
+        /**
+         * Provides access to the talon used in Venia's App UI component.
+         * @memberof TalonWrapperConfig
+         */
         App: {
+            /**
```
[the remark this answers is not in the document]
Comment (jcalcaben) [reply packages/peregrine/lib/targets/peregrine-intercept.js]: Alphabetized

File: packages/peregrine/lib/targets/peregrine-intercept.js (hunk 1 above)
Comment (jcalcaben) [reply packages/peregrine/lib/targets/peregrine-intercept.js]: Updated

File: packages/peregrine/lib/targets/TalonWrapperConfig.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+module.exports = TalonWrapperConfig;
+
+/**
+ * A particular Peregrine talon implementation which can be decorated with
+ * custom JavaScript before it is added to the PWA bundle and imported by React
+ * components.
+ */
+class WrappableTalon {
+    /** @hideconstructor */
+    constructor({ addTransform, talonFile }) {
```
Comment (zetlen) [inline packages/peregrine/lib/targets/TalonWrapperConfig.js:48]:
```suggestion
    constructor(addTransform, talonFile) {
```

Whoops! @jcalcaben Mistake here caused a regression that I didn't catch in the test suite. Please pull in this change or tests will fail :)

File: pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+### Requirements
+
+Wrapper modules for Peregrine talons must:
+- be implemented in a separate file from the build-time Target code
+- be accessible in frontend code via an `import`
+- be a valid ES Module
+- export one default function that implements the [TalonWrapper][] interface, i.e. it receives the original talon function as its parameter and must return a new talon function
+
+Extensions which use wrapper modules _must_ also intercept the [Buildpack `specialFeatures` target]({%link /pwa-buildpack/reference/targets/index.md %}#module_BuiltinTargets.specialFeatures) and set the feature flag `esModule: true`.
```
Comment (jcalcaben) [inline pwa-devdocs/src/peregrine/reference/targets/wrappable-talons/index.md:28]:
```suggestion
Extensions which use wrapper modules _must_ also intercept the [Buildpack `specialFeatures` target]({%link pwa-buildpack/reference/targets/index.md %}#module_BuiltinTargets.specialFeatures) and set the feature flag `esModule: true`.
```

**Pull request merged**